src/box/raft.c

```diff
@@ -261,24 +282,20 @@ static void
 box_raft_notify_have_quorum(void)
 {
 	struct raft *raft = box_raft();
+	bool has_healthy_quorum = replicaset_has_healthy_quorum();
 	switch (box_election_mode) {
 	case ELECTION_MODE_MANUAL:
 		/* Quorum loss shouldn't interfere with manual elections. */
-		/*
-		 * TODO: make a manually elected leader step off immediately due
-		 * to fencing.
-		 */
 		assert(!raft->is_cfg_candidate);
+		if (!has_healthy_quorum)
+			box_raft_fence();
 		break;
 	case ELECTION_MODE_CANDIDATE:
-		if (replicaset_has_healthy_quorum()) {
+		if (has_healthy_quorum) {
 			raft_cfg_is_candidate(raft, true);
 		} else if (raft->state == RAFT_STATE_CANDIDATE ||
 			   raft->state == RAFT_STATE_LEADER) {
-			/*
-			 * TODO: make the leader step off immediately due to
-			 * fencing.
-			 */
+			box_raft_fence();
 			raft_schedule_is_candidate(raft, false);
```

Comment:
Since you call `box_raft_fence` here, you may remove `raft_schedule_is_candidate`.
`box_raft_fence` supersedes it. It turns candidate off in current term and until further configuration.
`raft_schedule_is_candidate` will only turn candidate off in the next term.

Reply:
Fixed